packages/core/data-transfer/types/transfer-engine.d.ts

```diff
@@ -31,7 +34,7 @@ export interface ITransferEngine {
   /**
    * Start streaming selected data from the source to the destination
    */
-  transfer(): Promise<void>;
+  transfer(): Promise<void | ITransferResults<S, D>>;
```

Comment:
I think adding never will only complicate things since anything using the results will then have to check it's a result, but if we only throw or return results there's no need.

Reply:
Yep alright, you convinced me